easybuild/framework/easyconfig/tweak.py

```diff
@@ -119,8 +119,22 @@ def tweak(easyconfigs, build_specs, modtool, targetdirs=None):
             else:
                 target_toolchain['version'] = source_toolchain['version']
 
-            src_to_dst_tc_mapping = map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool)
+            try:
+                src_to_dst_tc_mapping = map_toolchain_hierarchies(source_toolchain, target_toolchain, modtool)
+            except EasyBuildError as err:
+                if build_option('force'):
```

Comment:
@bartoldeman We should probably check whether the raised exception is indeed the one raised by `match_minimum_tc_specs`, since (at least in theory) an `EasyBuildError` could be raised for a variety of reasons....

That could look something like this (untested):

```python
except EasyBuildError as err:
    # make sure exception was raised by match_minimum_tc_specs because toolchain mapping could not be done
    if "No possible mapping from source toolchain" in err.msg:

        if build_option('force'):
            warning_msg = "Combining --try-toolchain with --force for toolchains with unequal capabilities: "
            warning_msg += "disabling recursion and not changing (sub)toolchains for dependencies."
            print_warning(warning_msg)
            revert_to_regex = True
            modifying_toolchains = False
        else:
            error_msg = err.msg + '\n'
            error_msg += "Toolchain %s is not equivalent to toolchain %s in terms of capabilities. "
            error_msg += "(if you know what you are doing, you can use --force to proceed anyway)"
            raise EasyBuildError(error_msg, target_toolchain['name'], source_toolchain['name'])
    else:
        # simply re-raise the exception if something else went wrong
        raise err